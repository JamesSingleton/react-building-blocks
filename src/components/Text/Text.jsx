import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


export const SIZES = {
  tiny: 'tiny',
  small: 'small',
  medium: 'medium',
};

export const WEIGHTS = {
  thin: 'thin',
  normal: 'normal',
  bold: 'bold',
};

const StyledText = styled.span`
font-size: ${(props) => {
    if (props.size === SIZES.small) return '0.875rem';
    if (props.size === SIZES.tiny) return '0.75rem';
    return '1rem';
  }}
line-height: ${(props) => {
    if (props.size === SIZES.small) return '1.125rem';
    if (props.size === SIZES.tiny) return '0.9375rem';
    return '1.5rem';
  }}
  & > a {
    color: #3899EC;
    text-decoration: none;
    &:hover {
      color: #4EB7F5;
    }
  }
`;

const Text = ({
  tagName, size, secondary, weight, children,
}) => (
  <StyledText as={tagName} size={size} secondary={secondary} weight={weight}>
    {children}
  </StyledText>
);

Text.propTypes = {
  tagName: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.element,
  ]).isRequired,
  size: PropTypes.oneOf(Object.keys(SIZES)),
  secondary: PropTypes.bool,
  weight: PropTypes.oneOf(Object.keys(WEIGHTS)),
};

Text.defaultProps = {
  tagName: 'span',
  size: SIZES.medium,
  secondary: false,
  weight: WEIGHTS.thin,
};

export default Text;
