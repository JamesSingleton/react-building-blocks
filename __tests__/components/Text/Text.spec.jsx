import React from 'react';
import { mount } from 'enzyme';
import Text from '../../../src/components/Text';

describe('Text', () => {
  it('should render as a span', () => {
    const props = {
      tagName: 'span',
    };
    const renderedText = mount(<Text {...props}>Hello Span</Text>);
    expect(renderedText).toMatchSnapshot();
  });
  it('renders a small text component', () => {
    const props = {
      size: 'small',
    };
    const renderedText = mount(<Text {...props}>Hello Small Span</Text>);
    expect(renderedText).toMatchSnapshot();
  });
  it('renders a tiny text component', () => {
    const props = {
      size: 'tiny',
    };
    const renderedText = mount(<Text {...props}>Hello Tiny Span</Text>);
    expect(renderedText).toMatchSnapshot();
  });
});
