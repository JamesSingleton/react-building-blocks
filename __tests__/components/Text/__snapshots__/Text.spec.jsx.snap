// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders a small text component 1`] = `
<Text
  secondary={false}
  size="small"
  tagName="span"
  weight="thin"
>
  <styled.span
    as="span"
    secondary={false}
    size="small"
    weight="thin"
  >
    <StyledComponent
      as="span"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "cQhRnj",
            "rules": Array [
              "
font-size: ",
              [Function],
              "
line-height: ",
              [Function],
              "
  & > a {
    color: #3899EC;
    text-decoration: none;
    &:hover {
      color: #4EB7F5;
    }
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      secondary={false}
      size="small"
      weight="thin"
    >
      <span
        className="sc-bdVaJa cQhRnj"
        size="small"
      >
        Hello Small Span
      </span>
    </StyledComponent>
  </styled.span>
</Text>
`;

exports[`Text renders a tiny text component 1`] = `
<Text
  secondary={false}
  size="tiny"
  tagName="span"
  weight="thin"
>
  <styled.span
    as="span"
    secondary={false}
    size="tiny"
    weight="thin"
  >
    <StyledComponent
      as="span"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "hlDebU",
            "rules": Array [
              "
font-size: ",
              [Function],
              "
line-height: ",
              [Function],
              "
  & > a {
    color: #3899EC;
    text-decoration: none;
    &:hover {
      color: #4EB7F5;
    }
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      secondary={false}
      size="tiny"
      weight="thin"
    >
      <span
        className="sc-bdVaJa hlDebU"
        size="tiny"
      >
        Hello Tiny Span
      </span>
    </StyledComponent>
  </styled.span>
</Text>
`;

exports[`Text should render as a span 1`] = `
<Text
  secondary={false}
  size="medium"
  tagName="span"
  weight="thin"
>
  <styled.span
    as="span"
    secondary={false}
    size="medium"
    weight="thin"
  >
    <StyledComponent
      as="span"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "gIUtbY",
            "rules": Array [
              "
font-size: ",
              [Function],
              "
line-height: ",
              [Function],
              "
  & > a {
    color: #3899EC;
    text-decoration: none;
    &:hover {
      color: #4EB7F5;
    }
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      secondary={false}
      size="medium"
      weight="thin"
    >
      <span
        className="sc-bdVaJa gIUtbY"
        size="medium"
      >
        Hello Span
      </span>
    </StyledComponent>
  </styled.span>
</Text>
`;
